server/core/basic_cluster.go

```diff
@@ -238,10 +238,34 @@ func (bc *BasicCluster) GetAverageRegionSize() int64 {
 // PutStore put a store.
 func (bc *BasicCluster) PutStore(store *StoreInfo) {
 	bc.Lock()
+	defer bc.updateMaxScore(store)
 	defer bc.Unlock()
 	bc.Stores.SetStore(store)
 }
 
+func (bc *BasicCluster) updateMaxScore(store *StoreInfo) {
```

Comment:
maybe better to maintain max score in `bc.Stores`.

Reply:
but it is difficult to get opt in `bc.store`

Reply:
but `GetStores` return `[]*storeInfo` rather than `bc.Stores`